capplets/system-info/mate-system-info.c

```diff
@@ -0,0 +1,745 @@
+    }
+    windowing_system_text = get_windowing_system ();
+    label = g_object_get_data (G_OBJECT (info->windowing_system_row), "labelvalue");
+    set_lable_style (label, "gray", 12, windowing_system_text, FALSE);
+
+    de_text = get_mate_desktop_version ();
+    label = g_object_get_data (G_OBJECT (info->mate_version_row), "labelvalue");
+    set_lable_style (label, "gray", 12, de_text, FALSE);
+
+    os_type_text = get_os_type ();
+    label = g_object_get_data (G_OBJECT (info->os_type_row), "labelvalue");
+    set_lable_style (label, "gray", 12, os_type_text, FALSE);
+
+    os_name_text = get_os_name ();
+    label = g_object_get_data (G_OBJECT (info->os_name_row), "labelvalue");
+    set_lable_style (label, "gray", 12, os_name_text, FALSE);
+}
+
+static void
+mate_system_info_destroy (GtkWidget *widget)
+{
+    GTK_WIDGET_CLASS (mate_system_info_parent_class)->destroy (widget);
+}
+
+static void
+mate_system_info_class_init (MateSystemInfoClass *klass)
+{
+    GtkWidgetClass *widget_class = GTK_WIDGET_CLASS (klass);
+
+    widget_class->destroy = mate_system_info_destroy;
+    gtk_widget_class_set_template_from_resource (widget_class, "/org/mate/control-center/system-info/mate-system-info.ui");
+
+    gtk_widget_class_bind_template_child (widget_class, MateSystemInfo, hostname_row);
+    gtk_widget_class_bind_template_child (widget_class, MateSystemInfo, hardware_box);
+    gtk_widget_class_bind_template_child (widget_class, MateSystemInfo, disk_row);
+    gtk_widget_class_bind_template_child (widget_class, MateSystemInfo, mate_version_row);
+    gtk_widget_class_bind_template_child (widget_class, MateSystemInfo, graphics_row);
+    gtk_widget_class_bind_template_child (widget_class, MateSystemInfo, hardware_model_row);
+    gtk_widget_class_bind_template_child (widget_class, MateSystemInfo, memory_row);
+    gtk_widget_class_bind_template_child (widget_class, MateSystemInfo, os_box);
+    gtk_widget_class_bind_template_child (widget_class, MateSystemInfo, logo_image);
+    gtk_widget_class_bind_template_child (widget_class, MateSystemInfo, os_name_row);
+    gtk_widget_class_bind_template_child (widget_class, MateSystemInfo, os_type_row);
+    gtk_widget_class_bind_template_child (widget_class, MateSystemInfo, processor_row);
+    gtk_widget_class_bind_template_child (widget_class, MateSystemInfo, virtualization_row);
+    gtk_widget_class_bind_template_child (widget_class, MateSystemInfo, kernel_row);
+    gtk_widget_class_bind_template_child (widget_class, MateSystemInfo, windowing_system_row);
+}
+
+static void
+mate_system_info_init (MateSystemInfo *self)
+{
+    gtk_widget_init_template (GTK_WIDGET (self));
+    g_resources_register (mate_system_info_get_resource ());
+    mate_system_info_set_row (MATE_SYSTEM_INFO (self));
+}
+
+GtkWidget *
+mate_system_info_new (void)
+{
+    GtkWidget *dialog;
+
+    dialog = g_object_new (MATE_TYPE_SYSTEM_INFO,
+                           "use-header-bar", TRUE,
```

Comment:
Here is use-header-bar!

Reply:
```suggestion
                           "use-header-bar", TRUE,

    GtkWidget *dialog;

    dialog = g_object_new (MATE_TYPE_SYSTEM_INFO,
                           "use-header-bar", TRUE,
```

Reply:
I just tested


    dialog = g_object_new (MATE_TYPE_SYSTEM_INFO,
```suggestion
                           "use-header-bar", TRUE,
```
                           "use-header-bar", FALSE,

and as expected that gave us the normal SSD decorations

Reply:
dialog = g_object_new (MATE_TYPE_SYSTEM_INFO,
```suggestion
                           "use-header-bar", TRUE,
```
                           "use-header-bar", FALSE,

Reply:
I cannot figure out how to get the suggestions to work, but I just tested using
```

    dialog = g_object_new (MATE_TYPE_SYSTEM_INFO,
                           "use-header-bar", FALSE,
```
and that fixed the problem as expected and gave us the normal window decorations for MATE

Reply:
Determine whether to use header bar based on configuration

Reply:
Works fine in x11, but creates a problem in wayland, where we do not have an xsettings manager. A lot of dialogs there go to the GTK default, which is CSD. When running under wayland, we should unconditionally set
use_header_bar to FALSE or this dialog will be the only dialog from mate-control-center to get CSD decorations when compositors such as wayfire or sway are set to create server-side decorations. A compositor set to create client side decorations will be no difference of course.

In general, we should not use header bars anywhere in MATE, and should only allow them as a user option when this won't come with a side order of some users being forced to use them. If you want to make this user selectable in x11, we need to ensure we can get SSD decorations in wayland.